File: C++/plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int sz = digits.size();
        int carry = 1;
        vector<int> result;
        for (int i = 0; i < sz; i++) {
            int val = digits[sz - i - 1] + carry;
            carry = val / 10;
            result.push_back(val % 10);
        }
        if (carry) {
            result.push_back(carry);
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: Kotlin/plus-one.kt
class Solution {
    fun plusOne(digits: IntArray): IntArray {
        var n = digits.size-1
        var tmp: MutableList<Int> = digits.toMutableList()
        var carry = 1
        for (i in n downTo 0) {
            tmp[i] += carry
            carry = tmp[i] / 10
            tmp[i] %= 10
        }
        if (carry > 0) {
            tmp.add(0, carry)
        }
        return tmp.toIntArray()
    }
}